addons/website/static/src/snippets/s_table_of_content/000.js

```diff
@@ -42,7 +43,16 @@ const TableOfContent = publicWidget.Widget.extend({
         const $mainNavBar = $('#oe_main_menu_navbar');
         position += $mainNavBar.length ? $mainNavBar.outerHeight() : 0;
         position += isHorizontalNavbar ? this.$target.outerHeight() : 0;
-        $().getScrollingElement().scrollspy({target: '.s_table_of_content_navbar', method: 'offset', offset: position + 100, alwaysKeepFirstActive: true});
+        this.$scrollingElement = $().getScrollingElement();
+        if (this.previousPosition !== position) {
+            // If the scrollSpy already exists it must be destroyed before
+            // calling it again. Otherwise the call has no effect.
+            if (this.previousPosition > -1) {
+                this.$scrollingElement.scrollspy('dispose');
+            }
```

Comment:
If you just call dispose without checking and tracking `previousPosition`, does it not work? jQuery/bootstrap surely is smart enough to just do nothing if asked to dispose of something that does not exist.

Here I read this like I would read this:
```javascript
if (el.classList.contains('a')) {
    el.classList.remove('a');
}
```
The less code the better :wink:

Reply:
I removed the i`f (this.previousPosition > -1) {` condition. But I keep the `if (this.previousPosition !== position) {` condition because we don't want to call scrollspy when not needed.